pkg/controllers/awsloadbalancercontroller/webhooks.go

```diff
@@ -2,12 +2,371 @@ package awsloadbalancercontroller
+	if err != nil {
+		return fmt.Errorf("failed to set owner reference on desired ValidatingWebhookConfiguration %s: %w", desiredVWC.Name, err)
+	}
+
+	currentVWC, exists, err := r.currentValidatingWebhookConfiguration(ctx, desiredVWC.Name)
+	if err != nil {
+		return fmt.Errorf("failed to get current ValidatingWebhookConfiguration %q: %w", desiredVWC.Name, err)
+	}
+	if !exists {
+		err = r.Create(ctx, desiredVWC)
+		if err != nil {
+			return fmt.Errorf("failed to create ValidatingWebhookConfiguration %q: %w", desiredVWC.Name, err)
+		}
+	} else {
+		err = r.updateValidatingWebhookConfiguration(ctx, currentVWC, desiredVWC)
+		if err != nil {
+			return fmt.Errorf("failed to updated ValidatingWebhookConfiguration %s: %w", currentVWC.Name, err)
+		}
+	}
+
+	desiredMWC := desiredMutatingWebhookConfiguration(controller, service)
+	err = controllerutil.SetOwnerReference(controller, desiredMWC, r.Scheme)
+	if err != nil {
+		return fmt.Errorf("failed to set owner reference on desired MutatingWebhookConfiguration %s: %w", desiredMWC.Name, err)
+	}
+
+	currentMWC, exists, err := r.currentMutatingWebhookConfiguration(ctx, desiredMWC.Name)
+	if err != nil {
+		return fmt.Errorf("failed to get current MutatingWebhookConfiguration %q: %w", desiredMWC.Name, err)
+	}
+	if !exists {
+		err := r.Create(ctx, desiredMWC)
+		if err != nil {
+			return fmt.Errorf("failed to create MutatingWebhookConfiguration %q: %w", desiredMWC.Name, err)
+		}
+		return nil
+	}
+
+	err = r.updateMutatingWebhookConfiguration(ctx, currentMWC, desiredMWC)
+	if err != nil {
+		return fmt.Errorf("failed to updated ValidatingWebhookConfiguration %s: %w", currentVWC.Name, err)
+	}
 	return nil
 }
+
+func (r *AWSLoadBalancerControllerReconciler) currentValidatingWebhookConfiguration(ctx context.Context, name string) (*arv1.ValidatingWebhookConfiguration, bool, error) {
+	var currentVWC arv1.ValidatingWebhookConfiguration
+	err := r.Get(ctx, types.NamespacedName{Name: name}, &currentVWC)
+	if err != nil && errors.IsNotFound(err) {
+		return nil, false, nil
+	}
+	return &currentVWC, true, err
+}
+
+func (r *AWSLoadBalancerControllerReconciler) currentMutatingWebhookConfiguration(ctx context.Context, name string) (*arv1.MutatingWebhookConfiguration, bool, error) {
+	var currentMWC arv1.MutatingWebhookConfiguration
+	err := r.Get(ctx, types.NamespacedName{Name: name}, &currentMWC)
+	if err != nil && errors.IsNotFound(err) {
+		return nil, false, nil
+	}
+	return &currentMWC, true, nil
+}
+
+func desiredValidatingWebhookConfiguration(controller *albo.AWSLoadBalancerController, webhookService *corev1.Service) *arv1.ValidatingWebhookConfiguration {
+	return &arv1.ValidatingWebhookConfiguration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("%s-%s", controllerResourcePrefix, controller.Name),
+		},
+		Webhooks: []arv1.ValidatingWebhook{
+			{
+				Name: "vtargetgroupbinding.elbv2.k8s.aws",
+				ClientConfig: arv1.WebhookClientConfig{
```

Comment:
I don't see `caBundle` for the webhooks, the communication is plain http?
As we are not limited by the vanilla Kube support we can use OpenShift's [ca injector](https://docs.openshift.com/container-platform/4.10/security/certificates/service-serving-certificate.html#add-service-certificate-validating-webhook_service-serving-certificate) for the webhookConfigs and [serving cert secret](https://docs.openshift.com/container-platform/4.10/security/certificates/service-serving-certificate.html#add-service-certificate-validating-webhook_service-serving-certificate) for the service.

Reply:
Updated with the functionality mentioned.